kable-btleplug-ffi/src/peripheral_id.rs

```diff
@@ -0,0 +1,73 @@
+#[cfg(target_os = "windows")]
+use btleplug::api::BDAddr;
+#[cfg(target_os = "linux")]
+use serde::{Deserialize, Serialize};
+use std::fmt::Display;
+#[cfg(target_os = "windows")]
+use std::str::FromStr;
+
+#[derive(Eq, PartialEq, Hash, Clone, uniffi::Object)]
+#[uniffi::export(Display, Hash, Eq)]
+pub struct PeripheralId {
+    pub platform: btleplug::platform::PeripheralId,
+}
+
+#[cfg(target_os = "macos")]
+#[uniffi::export]
+impl PeripheralId {
+    #[uniffi::constructor]
+    fn new(value: String) -> Self {
+        PeripheralId {
```

Comment:
nit, unsure if more "rusty" or not to use `Self` in constructors? Asking because we have not been consistent here vs other structs in this module.
```suggestion
        Self {
```

Reply:
Good point. A quick search says `Self {` for the constructor might be more idiomatic.

Reply:
Same here re. a default-off lint. Enabled going forward.